tests/models/test_configuration.py

```diff
@@ -224,6 +227,32 @@ def test_relationships(self):
         self._db.commit()
         eq_([for_library.id], [x.id for x in library.settings])
 
+    def test_no_orphan_delete_cascade(self):
+        # Disconnecting a ConfigurationSetting from a Library or
+        # ExternalIntegration doesn't delete it, because it's fine for
+        # a ConfigurationSetting to have no associated Library or
+        # ExternalIntegration.
+
+        library = self._default_library
+        for_library = ConfigurationSetting.for_library(self._str, library)
+
+        integration = self._external_integration(self._str)
+        for_integration = ConfigurationSetting.for_externalintegration(
+            self._str, integration
+        )
+
+        # Remove library and external_integration.
+        for_library.library = None
+        for_integration.external_integration = None
+        self._db.commit()
+
+        # That was a weird thing to do, but the ConfigurationSettings
```

Comment:
Weird thing to do because this doesn't happen in the code, but can happen through the UI somehow?

Reply:
I don't think it happens at all. And I'd actually prefer the behavior where setting one of those fields to None immediately deletes the ConfigurationSetting. But "delete-orphan" isn't a good way to get that behavior in a situation where there's any legitimate reason for those fields to be None.

Reply:
I think it's weird to do that because the way I feel is, if you want to delete a database object, just delete it. Don't set a foreign key it depends on to a null value.